onnxruntime/core/framework/tensorprotoutils.cc

```diff
@@ -1092,10 +1137,66 @@ common::Status DenseTensorToSparseTensorProto(const ONNX_NAMESPACE::TensorProto&
                       IsZero<uint8_t>, CopyElement<uint8_t>, values, indices);
       break;
     }
+    case 2: {
+      // bytes
+      switch (data_type) {
+        case TensorProto_DataType_FLOAT16: {
+          SparsifyGeneric(dense_raw_data.get(), n_dense_elements, element_size,
+                          IsZero<BFloat16>, CopyElement<uint16_t>, values, indices);
+          break;
+        }
+        case TensorProto_DataType_BFLOAT16: {
+          SparsifyGeneric(dense_raw_data.get(), n_dense_elements, element_size,
+                          IsZero<MLFloat16>, CopyElement<uint16_t>, values, indices);
+          break;
+        }
+        case TensorProto_DataType_INT16:
+        case TensorProto_DataType_UINT16: {
+          SparsifyGeneric(dense_raw_data.get(), n_dense_elements, element_size,
+                          IsZero<uint16_t>, CopyElement<uint16_t>, values, indices);
+          break;
+        }
+        default:
+          ORT_THROW(false, "BUG! Report to onnxruntime team.");
+          break;
+      }
+      break;
+    }
     case 4: {
-      // float
-      SparsifyGeneric(dense_raw_data.get(), n_dense_elements, element_size,
-                      IsZero<uint32_t>, CopyElement<uint32_t>, values, indices);
+      switch (data_type) {
+        case TensorProto_DataType_FLOAT:
+          SparsifyGeneric(dense_raw_data.get(), n_dense_elements, element_size,
+                          IsZero<float>, CopyElement<uint32_t>, values, indices);
+          break;
+        case TensorProto_DataType_INT32:
+        case TensorProto_DataType_UINT32: {
+          SparsifyGeneric(dense_raw_data.get(), n_dense_elements, element_size,
+                          IsZero<uint32_t>, CopyElement<uint32_t>, values, indices);
+          break;
+        }
+        default:
+          ORT_THROW(false, "BUG! Report to onnxruntime team.");
+          break;
+      }
```

Comment:
I understand the MLFloat16 and BFloat16 need explicit special handling, but for anything else would it be good enough just to call SparsifyGeneric with any type that has the correct data size so we don't need these additional 'default:' cases for an unhandled type? We've already explicitly checked the type is in a known list via the previous usage of MLTypeCallDispatcher